Add org.salutatoi.Cagou

goffi-contrib opened the pull request:
Cagou is the desktop/mobiles frontend of Salut à Toi. Salut à Toi is an
advanced XMPP client offering features such as end-to-end encryption
(OTR + OMEMO), file sharing, remote controls, etc.

barthalion commented on org.salutatoi.Cagou.json:
```diff
@@ -0,0 +1,1463 @@
+{
+    "app-id": "org.salutatoi.Cagou",
+    "default-branch": "0.7.0",
```
It should be unset or `"branch": "stable"`.

goffi-contrib replied:
oh right, I'll change that thanks.

barthalion commented on org.salutatoi.Cagou.json:
```diff
@@ -0,0 +1,1463 @@
+        "--filesystem=home"
+    ],
+    "modules": [
+        {
```
Python 2 is available in https://github.com/flathub/shared-modules/

goffi-contrib replied:
I have seen indeed, but after doing that one unfortunately. I'll update to use the official one.

barthalion commented on org.salutatoi.Cagou.json:
```diff
@@ -0,0 +1,1463 @@
+            ]
+        },
+        {
+            "name": "libxslt",
```
All modules likely need some cleanups to delete development headers (etc.).

goffi-contrib replied:
I'll try to review non python modules to see what I can remove.

barthalion commented on org.salutatoi.Cagou.json:
```diff
@@ -0,0 +1,1463 @@
+            ]
+        },
+        {
+            "name": "setuptools",
```
Are all these modules shared between Primitivus and Jp? I would consider keeping them in separate file and included as git submodule to ease maintenance.

goffi-contrib replied:
yes `Primitivus` and `jp` share most of the dependencies (there are some more for Primitivus, notably `urwid`). But maintenance is really easy on my side, I've made a script to generate the manifest with the right dependencies in the right order automatically, and it is using a template so I can tell the script to change anything specific. I can maintain things easily like this and using git submodules would in fact complicate my workflow.

barthalion replied:
Alright, you're the captain here!

barthalion commented on org.salutatoi.Cagou.appdata.xml:
```diff
@@ -0,0 +1,58 @@
+<?xml version='1.0' encoding='utf-8'?>
```
Desktop and appdata files ideally should be part of upstream repositories and installed by project buildsystem if possible – other distributions are also using them.

goffi-contrib replied:
Indeed. Next version will be Python 3 only and I've done many changes already, I'll add those script upstream for the next version, but I rather keep them here for this 0.7.0 version.

barthalion commented:
bot, build org.salutatoi.Cagou

goffi-contrib commented:
Hello, thank for your review. I'll address the comments individually and make an update as soon as possible (quite busy right now).

goffi-contrib commented:
bot, build org.salutatoi.Cagou

goffi-contrib commented:
@barthalion following your review, I've removed the `branch`, used flathub's shared-modules for `python2`, and removed `libxslt` after realizing it was in freedesktop runtime. I've also updated appdata file with `content_rating` as it was making the first test build fail.

I'm not sure why build are failing on non x86_64 platforms, `Pillow` complains that `The headers or library files could not be found for zlib` but zlib should be in freedesktop SDK right?

Let me know if there is anything else to change, thanks you :)

nedrichards commented:
bot, build org.salutatoi.Cagou

LeandroStanger commented:
bot, build org.salutatoi.Cagou
